.github/workflows/ui.yaml

```
@@ -159,3 +159,41 @@ jobs:
           path: |
             ui/cypress/debug
             debug
+
+  sync-lockfile:
+    name: Sync pnpm-lock.yaml for Snyk PRs
+    if: contains(github.event.pull_request.title, '[Snyk]')
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.CI_TOKEN }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+
+      - name: Install PNPM v9
+        run: npm install -g pnpm@9
+
+      - name: Update pnpm-lock.yaml
+        run: pnpm --prefix ui install --no-frozen-lockfile
```

What about using `--lockfile-only` instead?
https://pnpm.io/cli/install#--lockfile-only

```suggestion
        run: pnpm --prefix ui install --lockfile-only
```

Would that work? Have you tried this?